demo/lib/vm.bash

```diff
@@ -455,25 +455,144 @@ vm-install-cri-resmgr() {
+        cat <<EOF | vm-pipe-to-file webhook/install-cfssl.sh
+VERSION=\$(curl --silent "https://api.github.com/repos/cloudflare/cfssl/releases/latest" | grep '"tag_name"' | sed -E 's/.*"([^"]+)".*/\1/')
+VNUMBER=\${VERSION#"v"}
+wget -q https://github.com/cloudflare/cfssl/releases/download/\${VERSION}/cfssl_\${VNUMBER}_linux_amd64 -O cfssl
+chmod +x cfssl
+mv cfssl /usr/local/bin
+EOF
+        vm-command "chmod a+x webhook/install-cfssl.sh; webhook/install-cfssl.sh"
+    }
+    vm-command-q "command -v jq" >/dev/null || {
+        vm-install-pkg jq
+    }
+    # Create webhook certificate
+    cat <<EOF | vm-pipe-to-file webhook/csr-config.json
+{
+    "CN": "${service}.${namespace}.svc",
+    "hosts": [
+        "${service}",
+        "${service}.${namespace}",
+        "${service}.${namespace}.svc"
+    ],
+    "key": {
+        "algo": "rsa",
+        "size": 2048
+    }
+}
+EOF
+    vm-command "cfssl genkey -loglevel=2 webhook/csr-config.json > webhook/csr.json"
+    vm-command "jq --raw-output '.key' webhook/csr.json > webhook/server-key.pem"
+    vm-command "jq --raw-output '.csr' webhook/csr.json > webhook/server.csr"
+    local server_csr_b64
+    server_csr_b64="$(vm-command-q "cat webhook/server.csr" | base64 -w 0)"
+    cat <<EOF | vm-pipe-to-file webhook/csr.yaml
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: ${service}.${namespace}
+spec:
+  request: ${server_csr_b64}
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+EOF
+    kubectl create -f webhook/csr.yaml
+    vm-run-until "kubectl get csr | grep -q  ${service}.${namespace}" ||
+        error "creating certificate signing request failed"
+    kubectl certificate approve "${service}.${namespace}" ||
+        error "approving certificate signing request failed"
+    vm-run-until "kubectl get csr | grep ${service}.${namespace} | grep Issued" ||
+        error "certificate signing request not Issued after approval"
+    vm-command "kubectl get csr ${service}.${namespace} -o jsonpath='{.status.certificate}' | base64 -d > webhook/server-crt.pem"
+    # Allow webhook to run on node tainted by cmk=true
+    sed -e "s|IMAGE_PLACEHOLDER|$webhook_image_repotag|" \
+        -e 's|^\(\s*\)tolerations:$|\1tolerations:\n\1  - {\"key\": \"cmk\", \"operator\": \"Equal\", \"value\": \"true\", \"effect\": \"NoSchedule\"}|g' \
+        -e 's/imagePullPolicy: Always/imagePullPolicy: Never/' \
+        < "${HOST_PROJECT_DIR}/cmd/cri-resmgr-webhook/webhook-deployment.yaml" \
+        | vm-pipe-to-file webhook/webhook-deployment.yaml
+    # Create secret that contains svc.crt and svc.key for webhook deployment
+    local server_crt_b64 server_key_b64
+    server_crt_b64="$(vm-command-q "cat webhook/server-crt.pem" | base64 -w 0)"
+    server_key_b64="$(vm-command-q "cat webhook/server-key.pem" | base64 -w 0)"
+    cat <<EOF | vm-pipe-to-file --append webhook/webhook-deployment.yaml
```

Comment:
Do we really need to create certificates through kubernetes? Could/would it be simpler just to create a self-signed ca and crt/key pair signed by that with ssl?

Reply:
No we don't. How happy can one be when replacing a code like that with pretty much a single line... :) :) :) openssl 1.1.1 rocks.

Maybe we could have the openssl magic in our user guide, WDYT?

Reply:
> Maybe we could have the openssl magic in our user guide, WDYT?

Mm, might be useful, agree